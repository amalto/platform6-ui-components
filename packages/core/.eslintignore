dist/
loader/

src/components.d.ts