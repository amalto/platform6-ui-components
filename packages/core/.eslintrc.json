{
  "extends": ["../../.eslintrc.json", "plugin:@stencil/recommended", "plugin:jest/recommended", "plugin:jsx-a11y/recommended"],
  "parserOptions": {
    "project": ["./tsconfig.json", "./tsconfig.lint.json"],
    "ecmaFeatures": {
      "jsx": true // Allows for the parsing of JSX
    }
  },
  "plugins": ["jest"],

  "rules": {
    "jsx-a11y/label-has-associated-control": [
      2,
      {
        "assert": "either"
      }
    ],
    "react/jsx-props-no-spreading": "off",
    "react/no-unknown-property": [2, { "ignore": ["class"] }]
  },
  "settings": {
    "react": {
      "pragma": "h"
    }
  }
}
