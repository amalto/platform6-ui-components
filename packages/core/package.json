{
  "name": "@platform6/components",
  "version": "0.0.1",
  "description": "Platform 6 UI components",
  "repository": {
    "type": "git",
    "url": "https://github.com/amalto/platform6-ui-components"
  },
  "license": "MIT",
  "main": "dist/index.cjs.js",
  "module": "dist/custom-elements/index.js",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "es2015": "dist/esm/index.mjs",
  "es2017": "dist/esm/index.mjs",
  "unpkg": "dist/stencil-component/stencil-component.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/",
    "loader/"
  ],
  "scripts": {
    "build": "stencil build --docs",
    "format": "prettier --write ./",
    "postformat": "npm run lint.fix",
    "generate": "stencil generate",
    "lint": "npm run lint.ts && npm run lint.sass",
    "lint.fix": "npm run lint.ts.fix && npm run lint.sass.fix",
    "lint.sass": "stylelint 'src/**/*.scss'",
    "lint.sass.fix": "npm run lint.sass -- --fix",
    "lint.ts": "eslint --cache --ext .js,.jsx,.ts,.tsx ./",
    "lint.ts.fix": "npm run lint.ts -- --fix",
    "start": "stencil build --docs --dev --watch --serve",
    "test": "stencil test --spec --e2e",
    "test.watch": "stencil test --spec --e2e --watchAll",
    "storybook": "start-storybook -p 6006 -s ./dist --ci",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "@creativebulma/bulma-tooltip": "^1.2.0",
    "@fortawesome/fontawesome-svg-core": "^1.2.32",
    "@fortawesome/free-brands-svg-icons": "^5.15.1",
    "@fortawesome/free-regular-svg-icons": "^5.15.1",
    "@fortawesome/free-solid-svg-icons": "^5.15.1",
    "@stencil/core": "2.2.0",
    "bulma": "^0.9.1",
    "bulma-calendar": "^6.0.9",
    "choices.js": "9.0.1",
    "lit-html": "^1.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@stencil/eslint-plugin": "^0.3.1",
    "@stencil/postcss": "^1.0.1",
    "@stencil/react-output-target": "0.0.9",
    "@stencil/sass": "^1.3.2",
    "@storybook/addon-a11y": "^6.0.28",
    "@storybook/addon-actions": "^6.0.28",
    "@storybook/addon-essentials": "^6.0.28",
    "@storybook/addon-links": "^6.0.28",
    "@storybook/html": "^6.0.28",
    "@storybook/preset-scss": "^1.0.3",
    "@storybook/theming": "^6.0.28",
    "@storybook/web-components": "^6.0.28",
    "@types/jest": "^26.0.15",
    "@types/js-beautify": "^1.11.0",
    "@typescript-eslint/eslint-plugin": "^4.6.1",
    "@typescript-eslint/parser": "^4.6.1",
    "autoprefixer": "^9.8.6",
    "babel-loader": "^8.1.0",
    "css-loader": "^5.0.0",
    "eslint-plugin-jest": "^24.1.0",
    "jest": "^26.6.3",
    "jest-cli": "^26.6.3",
    "js-beautify": "^1.13.0",
    "prettier": "^2.1.2",
    "puppeteer": "^5.4.1",
    "react-is": "^17.0.1",
    "sass-loader": "^10.0.5",
    "style-loader": "^2.0.0",
    "typescript": "4.0.5"
  }
}
