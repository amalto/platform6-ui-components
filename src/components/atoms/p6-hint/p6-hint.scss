@import "bulma/sass/form/tools";

:host {
  @extend .help;
}

@each $name, $pair in $colors {
  $color: nth($pair, 1);
  :host(.is-#{$name}) {
    color: $color;
  }
}
