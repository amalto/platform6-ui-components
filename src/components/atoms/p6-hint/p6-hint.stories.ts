import { Components } from "../../../components";
import {
  getElement,
  makeStory,
  ModeStoryMaker,
  Props,
} from "../../../shared/storybook/stories";
import { Mode } from "../../../shared/types";

const component = "p6-hint";

export default {
  title: "Atoms/Hint",
  component,
};

const getStoryField = (
  text: string,
  props?: Props<Components.P6Hint>
): HTMLElement => getElement(component, text, props);

// Default
export const Default = makeStory<{ label: string; mode: Mode }>({
  args: { label: "hint", mode: Mode.success },
  builder: ({ label, ...props }): HTMLElement =>
    getStoryField(label, { ...props }),
});

// --- Mode
export const Modes = makeStory({
  builder: (): HTMLElement =>
    ModeStoryMaker(({ key, value }) =>
      getStoryField(key, {
        mode: value,
      })
    ),
});
