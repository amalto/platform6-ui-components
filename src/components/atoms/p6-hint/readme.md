# p6-hint

<!-- Auto Generated Below -->

## Properties

| Property | Attribute | Description              | Type                                                                     | Default     |
| -------- | --------- | ------------------------ | ------------------------------------------------------------------------ | ----------- |
| `mode`   | `mode`    | set the mode of the hint | `"danger" \| "default" \| "info" \| "primary" \| "success" \| "warning"` | `"default"` |

---

_Built with [StencilJS](https://stenciljs.com/)_
