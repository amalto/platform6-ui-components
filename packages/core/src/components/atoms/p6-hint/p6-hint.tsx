import { Component, ComponentInterface, h, Host, JSX, Prop } from '@stencil/core';
import { Mode } from '../../../shared/types';
import { getModeClass } from '../../../utils/classes';

@Component({
  tag: 'p6-hint',
  styleUrl: 'p6-hint.scss',
  shadow: true,
})
export class P6Hint implements ComponentInterface {
  /**
   * set the mode of the hint
   */
  @Prop() mode: Mode = Mode.default;

  render(): JSX.Element {
    return (
      <Host
        class={{
          ...getModeClass(this.mode),
        }}
      >
        <slot />
      </Host>
    );
  }
}
