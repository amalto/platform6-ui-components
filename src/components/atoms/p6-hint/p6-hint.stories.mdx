import { library } from "@fortawesome/fontawesome-svg-core";
import { fab } from "@fortawesome/free-brands-svg-icons";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { action } from "@storybook/addon-actions";
import { Meta, Preview, Story } from "@storybook/addon-docs/blocks";
import { select, text, boolean, withKnobs } from "@storybook/addon-knobs";
import readme from "./readme.md";

<Meta title="Atoms/Hint" component="p6-hint" parameters={{ notes: readme }} />

# Hint

<Preview>
  <Story name="Default" height="42px">
    {`
      <P6-hint>default</P6-hint>
    `}
  </Story>
</Preview>

## Mode

<Preview>
  <Story name="Mode" height="168px">
    {`
      <P6-hint mode="danger">a hint message with the mode \`danger\`</P6-hint>
      <P6-hint mode="warning">a hint message with the mode \`warning\`</P6-hint>
      <P6-hint mode="default">a hint message without mode or set to \`default\`</P6-hint>
      <P6-hint mode="info">a hint message with the mode \`info\`</P6-hint>
      <P6-hint mode="success">a hint message with the mode \`success\`</P6-hint>
      <P6-hint mode="primary">a hint message with the mode \`primary\`</P6-hint>
    `}
  </Story>
</Preview>
