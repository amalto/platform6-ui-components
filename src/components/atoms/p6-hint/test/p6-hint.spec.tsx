import { newSpecPage } from "@stencil/core/testing";
import { P6Hint } from "../p6-hint";

describe("p6-hint", () => {
  it("renders", async () => {
    const page = await newSpecPage({
      components: [P6Hint],
      html: `<p6-hint>hint text</p6-hint>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  describe("mode", () => {
    const modes = ["danger", "warning", "info", "success", "primary"];

    it.each(modes)('has the class "is-%s" when mode is "%s"', async (mode) => {
      const page = await newSpecPage({
        components: [P6Hint],
        html: `<p6-hint mode=${mode}>${mode}</p6-hint>`,
      });
      expect(page.root).toMatchSnapshot();
    });

    it("does not add class when mode is default", async () => {
      const page = await newSpecPage({
        components: [P6Hint],
        html: `<p6-hint mode="default">default</p6-hint>`,
      });
      expect(page.root).toMatchSnapshot();
    });

    it("does not add class when mode is undefined", async () => {
      const page = await newSpecPage({
        components: [P6Hint],
        html: `<p6-hint mode=${undefined}>undefined</p6-hint>`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });
});
