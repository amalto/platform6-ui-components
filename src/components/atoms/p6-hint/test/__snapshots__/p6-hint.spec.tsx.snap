// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-hint mode does not add class when mode is default 1`] = `
<p6-hint mode="default">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  default
</p6-hint>
`;

exports[`p6-hint mode does not add class when mode is undefined 1`] = `
<p6-hint mode="undefined">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  undefined
</p6-hint>
`;

exports[`p6-hint mode has the class "is-danger" when mode is "%s" 1`] = `
<p6-hint class="is-danger" mode="danger">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  danger
</p6-hint>
`;

exports[`p6-hint mode has the class "is-info" when mode is "%s" 1`] = `
<p6-hint class="is-info" mode="info">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  info
</p6-hint>
`;

exports[`p6-hint mode has the class "is-primary" when mode is "%s" 1`] = `
<p6-hint class="is-primary" mode="primary">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  primary
</p6-hint>
`;

exports[`p6-hint mode has the class "is-success" when mode is "%s" 1`] = `
<p6-hint class="is-success" mode="success">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  success
</p6-hint>
`;

exports[`p6-hint mode has the class "is-warning" when mode is "%s" 1`] = `
<p6-hint class="is-warning" mode="warning">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  warning
</p6-hint>
`;

exports[`p6-hint renders 1`] = `
<p6-hint>
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  hint text
</p6-hint>
`;
